pkg/cableengine/syncer/syncer.go

```diff
@@ -169,7 +169,7 @@ func (i *GatewaySyncer) generateGatewayObject() (*v1.Gateway, error) {
 	}
 
 	klog.V(log.TRACE).Infof("Generated Gateway object: %+v", gateway)
-	return &gateway, nil
+	return &gateway, err
```

Comment:
We want to return nil here - we only use `err` to set `StatusFailure` on L162. This causes a unit test to fail (yay for regression tests 😄 ).  We could ignore the linting error or  set `err` to nil after L163 to silence the error. However since we now always return nil error, we might as well remove it from the return signature. If any error occurred generating the object, we'd want to report that in the `StatusFailure` anyway.

Reply:
Done in the latest, thanks @tpantelis!